dotcom-rendering/src/lib/discussionApi.tsx

```diff
@@ -436,3 +438,42 @@ export const getMoreResponses = async (
 
 	return parseCommentRepliesResponse(jsonResult.value);
 };
+
+const buildParams = (filters: FilterOptions): URLSearchParams => {
+	return new URLSearchParams({
+		// Frontend uses the 'recommendations' key to store this options but the api expects
+		// 'mostRecommended' so we have to map here to support both
+		orderBy:
+			filters.orderBy === 'recommendations'
+				? 'mostRecommended'
+				: filters.orderBy,
+		pageSize: String(filters.pageSize),
+		displayThreaded: String(
+			filters.threads === 'collapsed' || filters.threads === 'expanded',
+		),
+	});
+};
+
+export const getCommentContext = async (
+	ajaxUrl: string,
+	commentId: number,
+	filters: FilterOptions,
+): Promise<Result<GetDiscussionError, CommentContextType>> => {
+	const url = joinUrl(ajaxUrl, 'comment', commentId.toString(), 'context');
+	const params = buildParams(filters);
+
+	const jsonResult = await fetchJSON(url + '?' + params.toString());
+
+	if (jsonResult.kind === 'error') return jsonResult;
+
+	const result = safeParse(getCommentContextResponseSchema, jsonResult.value);
+
+	if (!result.success) {
+		return error('ParsingError');
+	}
+	if (result.output.status !== 'ok') {
+		return error('ApiError');
+	}
```

Comment:
I think that this is an impossible branch, and ESLint shoud be able to identify it.